torch/quantization/_correct_bias.py

```diff
@@ -0,0 +1,292 @@
+    with torch.no_grad():
+        quantized_model(stack)
+
+    remove_hooks(quantized_model)
+    remove_shadow(quantized_model)
+
+def parallel_bias_correction(float_model, quantized_model, img_data, white_list=_supported_modules):
+    ''' Applies bias correction on the whitelisted submodules of the quantized model
+    MeanLoggers are added to float and quantized models to record the running means of
+    their inputs and outputs, after running all the data provided by img_data, the results
+    from the loggers are fed into the bias correction function
+    '''
+
+    qconfig_debug = torch.quantization.QConfig(activation=MeanLogger, weight=None)
+    float_model.qconfig = qconfig_debug
+    quantized_model.qconfig = qconfig_debug
+
+    torch.quantization.prepare(float_model, inplace=True, white_list=white_list, prehook=MeanLogger)
+    torch.quantization.prepare(quantized_model, inplace=True, white_list=white_list,
+                                observer_non_leaf_module_list=[nnq.Linear], prehook=MeanLogger)
+    batch_size = None
+    # batch size is used here to avoid an adding error in the MeanLogger due
+    # to the last batch of the dataset possibly being a different size than
+    # the previous batches
+    for data in img_data:
+        with torch.no_grad():
+            if batch_size is None:
+                batch_size = data[0].size(0)  # getting batch size
+            if data[0].size(0) == batch_size:
+                float_model(data[0])
+                quantized_model(data[0])
+
+
+    output_logger, input_logger = get_inputs_n_outputs(float_model, quantized_model)
+    correct_quantized_bias(output_logger, input_logger, float_model, quantized_model)
+
+def correct_quantized_bias(expected_output, expected_input, float_model, quantized_model):
+    ''' Inplace function to modify the weights of the quantized_model based off the
+    recorded differences in weights provided by output_dict
+
+    '''
+    for key in expected_output:
+        # submodules associated with key
+        float_submodule = get_module(float_model, key[:-1])
+        quantized_submodule = get_module(quantized_model, key[:-1])
+
+        # checking for existence of bias attribute
+        if hasattr(quantized_submodule, 'bias') and type(quantized_submodule) in _supported_modules:
+            if (isinstance(quantized_submodule.bias, torch.nn.parameter.Parameter) and (quantized_submodule.bias is not None)) or \
+                    (quantized_submodule.bias() is not None):
+
+                bias = get_param(quantized_submodule, 'bias')
+
+                # grabbing the weights
+                float_weight = get_param(float_submodule, 'weight')
+                quantized_weight = get_param(quantized_submodule, 'weight')
+                if quantized_weight.is_quantized:
+                    quantized_weight = quantized_weight.dequantize()
+
+                error_matrix = float_weight - quantized_weight
```

Comment:
The logic here is incorrect. Please start with the simpler flow: use the logger to compute the mean of the difference in activations between the float and quantized module, update the bias with this difference. Lets do experiments to see if the same logic works for conv and linear